src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,165 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.Scanner;
+
+/**
+ * The Storage program implements an application which saves and updates a todo list file.
+ *
+ * @author joey-chance
+ * @version 1.0
+ * @since 2022-02-05
+ */
+public class Storage {
+    protected File directory;
+    protected File file;
+
+    /**
+     * Returns a Storage object which saves and updates a todo list file.
+     *
+     * @param directoryPath path to the directory which the todo list is stored
+     * @param filePath path to the todo list file
+     */
+    public Storage(String directoryPath, String filePath) {
+        this.directory = new File(directoryPath);
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Loads the todo list stored in the file to a LinkedList.
+     *
+     * @return LinkedList containing all of the tasks from the todo list file
+     */
+    public LinkedList<Task> load() {
+        LinkedList<Task> tasks = new LinkedList<>();
+        try {
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+            if (!file.exists()) {
+                file.createNewFile();
+                return tasks;
+            }
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String type = scanner.next();
+                String status = scanner.next();
+                String description = scanner.nextLine().strip();
+                switch (type) {
```

Comment:
Perhaps you can consider adding a blank line here to separate the parsing code block and the matching code block :)